ecosystem/sep-0006.md

```diff
@@ -129,14 +132,25 @@ Examples:
 }
 ```
 
+##### Special Cases
+###### Stellar account does not exist
 
 If the given Stellar `account` does not exist, on receipt of the deposit, the anchor should use `create_account` to create the account with at least enough XLM for the minimum reserve and a trust line (2.01 XLM is recommended). The anchor should take some of the asset that is sent in to pay for the XLM. The anchor should not list this minimal funding as a fee because the user still receives the value of the XLM in their account. The anchor should detect if the account has been created before returning deposit information, and adjust the `min_amount` in the response to be at least the amount needed to create the account.
 
 Since the anchor doesn't have the user account's secret key, the user must create a trust line to the anchor's asset before the anchor can send the remaining asset tokens to the user's account. The anchor should listen for the user to establish this trust line. Once the trust line is there, the anchor should send the remaining asset tokens to the account in Stellar to complete the deposit.
 
 If the anchor does not support creating new accounts for users and `account` doesn't exist yet, the anchor should return a `400 Bad Request` error. The response body should be a JSON object containing an `error` field that explains why the request failed.
 
-In the case where the account has already been created, the wallet or client should ensure that the user has already trusted the asset before initiating the deposit flow.
+###### Stellar account doesn't trust asset
+
+The deposit flow can only be fulfilled if the Stellar `Account` has established a trust line for the given asset. To ensure this is accomplished, when initiating the deposit flow, `Wallet` should check if the `Account` has a trust line for the given asset. If it doesn't:
+
+1. `Wallet` checks if `Account` has enough funds to create a trust line. If it does, skip to step `4`.
+2. If `Account` doesn't have enough funds, `Wallet` starts listening for transactions to the given `Account`, waiting for it to have enough funds for a trust line.
+3. When asked for a deposit, `Anchor` detects if `Account` has enough XLM to create a trust line. If it doesn't, `Anchor` sends the needed amount of XLM to the `Account` for creating a trust line. `Issuer` then starts listening for trust line creations for that `Account`.
+4. `Wallet` detects the arrival of funds in the `Account`, and establishes a trust line.
+5. `Anchor` detects the trust line creation in the `Account`. If the asset is `AUTH_REQUIRED`, `Anchor` either approves the new trust line, or returns any of the `customer_info_needed` responses until KYC conditions are met.
```

Comment:
I believe both of these special cases only come up after the issuer has received the deposit. The issuer needs to have KYCed the user before giving deposit instructions, so at the stage that the issuer is handling the "stellar account doesn't trust asset case", the user will already have satisfied KYC requirements. As a result, the issuer should be able to immediately approve the new trustline and does not need to give any further `customer_info_needed` responses.

Reply:
Yes, that makes sense – I've pushed some changes to reflect this behavior.